src/ucp/proto/proto_init.c

```diff
@@ -189,13 +189,13 @@ ucp_proto_perf_envelope_make(const ucp_proto_perf_list_t *perf_list,
 }
 
 ucs_status_t
-ucp_proto_init_parallel_stages(const ucp_proto_common_init_params_t *params,
+ucp_proto_init_parallel_stages(const ucp_proto_init_params_t *params,
                                size_t range_start, size_t range_end,
                                size_t frag_size, double bias,
                                const ucp_proto_perf_range_t **stages,
-                               unsigned num_stages)
+                               unsigned num_stages, unsigned flags)
```

Comment:
can we pass frag_size==SIZE_MAX when need to pass flags with UCP_PROTO_COMMON_INIT_FLAG_SINGLE_FRAG? to avoid adding extra parameter

Reply:
It seems like we can, but I think that passing additional argument is less evil then passing `frag_size` that is higher then `max_length` in some cases. It would bring additional complexity to `ucp_proto_common_init_caps` and `ucp_proto_init_parallel_stages` logic.